circleci: failed tests: TestReplicaIDChangePending

The following test appears to have failed:

[#17936](https://circleci.com/gh/cockroachdb/cockroach/17936):

```
I160517 02:01:16.343789 storage/replica.go:2157  [node=1,store=1,range=1] gossiping cluster id {"0f2d0ca0-16a9-4296-8ed6-e86de17fc333"} from store 1, range 1
I160517 02:01:16.345312 storage/replica_command.go:1470  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 1s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I160517 02:01:16.346295 storage/replica.go:2166  [node=1,store=1,range=1] gossiping sentinel from store 1, range 1
I160517 02:01:16.346567 storage/replica.go:2172  [node=1,store=1,range=1] gossiping first range from store 1, range 1
I160517 02:01:16.350779 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- FAIL: TestReplicaIDChangePending (0.02s)
	replica_test.go:5580: command was not re-proposed
=== RUN   TestScannerAddToQueues
--- PASS: TestScannerAddToQueues (0.01s)
=== RUN   TestScannerTiming
I160517 02:01:16.468857 storage/scanner_test.go:257  0: average scan: 15.504706ms
I160517 02:01:16.569583 storage/scanner_test.go:257  1: average scan: 25.395355ms
--- PASS: TestScannerTiming (0.21s)
=== RUN   TestScannerPaceInterval
--- PASS: TestScannerPaceInterval (0.01s)
=== RUN   TestScannerEmptyRangeSet
--
=== RUN   Example_rebalancing
W160517 02:02:07.099029 gossip/gossip.go:893  not connected to cluster; use --join to specify a connected node
I160517 02:02:07.960286 stopper.go:352  draining; tasks left:
19     gossip/infostore.go:286
--- PASS: Example_rebalancing (0.86s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	57.407s
=== RUN   TestBatchBasics
I160517 02:02:22.596293 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160517 02:02:22.598438 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.03s)
=== RUN   TestBatchRepr
I160517 02:02:22.653736 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160517 02:02:22.655749 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.04s)
=== RUN   TestBatchGet
I160517 02:02:22.663495 storage/engine/rocksdb.go:137  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -5544,6 +5544,11 @@ func TestReplicaIDChangePending(t *testing.T) {
 	defer tc.Stop()
 	rng := tc.rng
 
+	// Stop the command from being proposed to the raft group and being removed.
+	rng.mu.Lock()
+	rng.mu.proposeRaftCommandFn = func(p *pendingCmd) error { return nil }
+	rng.mu.Unlock()
+
 	// Add a command to the pending list
 	ba := roachpb.BatchRequest{}
 	ba.Timestamp = tc.clock.Now()

